cpusupport/cpusupport_x86_shani.c

```diff
@@ -0,0 +1,29 @@
+#include "cpusupport.h"
+
+#ifdef CPUSUPPORT_X86_CPUID
+#include <cpuid.h>
+
+#define CPUID_SHANI_BIT (1 << 29)
+#endif
+
+CPUSUPPORT_FEATURE_DECL(x86, shani)
+{
+#ifdef CPUSUPPORT_X86_CPUID
+	unsigned int eax, ebx, ecx, edx;
+
+	/* Check if CPUID supports the level we need. */
+	if (!__get_cpuid(0, &eax, &ebx, &ecx, &edx))
+		goto unsupported;
+	if (eax < 7)
+		goto unsupported;
+
+	/* Ask about CPU features.  This macro takes e?x as pointers! */
+	__cpuid_count(7, 0, eax, ebx, ecx, edx);
```

Comment:
Why not `__get_cpuid` like we do for the other cpusupport functions?

Reply:
Oh, just saw the commit about this, never mind.

Reply:
We should add `__cpuid_count` to the test in `cpusupport-X86-CPUID.c` in case there are platforms which have `__get_cpuid` but not `__cpuid_count`.